src/java.desktop/share/classes/sun/swing/text/TextComponentPrintable.java

```diff
@@ -340,9 +339,9 @@ private JTextComponent createPrintShellOnEDT(final JTextComponent textComponent)
                     }
                     @Override
                     public FontMetrics getFontMetrics(Font font) {
-                        return (frc.get() == null)
+                        return (frc == null)
                             ? super.getFontMetrics(font)
```

Comment:
Should the be set to local variable to avoid 2 volatile reads?

Reply:
I agree, it's better to read once.